src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_tts_language_change_notification.cc

```diff
@@ -88,11 +88,9 @@ void OnTTSLanguageChangeNotification::Run() {
   (*message_)[strings::params][strings::function_id] =
       static_cast<int32_t>(mobile_apis::FunctionID::OnLanguageChangeID);
 
-  const ApplicationSet& accessor =
+  const ApplicationSet& applications =
       application_manager_.applications().GetData();
-  ApplicationSetIt it = accessor.begin();
-  for (; accessor.end() != it;) {
-    ApplicationSharedPtr app = *it++;
+  for (auto app : applications) {
```

Comment:
I believe this didn't use a foreach loop for a reason. It looks like this loop can unregister applications, making the iterator invalid.
```suggestion
  ApplicationSetIt it = accessor.begin();
  for (; accessor.end() != it;) {
    ApplicationSharedPtr app = *it;
    ++it;
```

Reply:
I don't understand the functional difference between

```
for (auto app : applications)
vs
for (auto it = applications.begin(); it != applications.end(); ++it)
```

Am I misunderstanding your suggestion? If the iterator is invalidated I think both statements would be affected the same.

Regardless of the above, I thought this iterator is of the `DataAccessor::GetData`, a copy of the `ApplicationSet`. While `UnregisterApplication` does remove an app from the set `ApplicationManagerImpl::applications_,` it doesn't invalidate the iterator of this copy, and that's why we don't have to hold the lock on `applications_list_lock_ptr_` during the loop also.

Reply:
Yeah, to be honest it might be fine as-is, but considering that this is already handling this is in a safer manner, I would rather not change it. Invalid iterator issues can be quite time-consuming to debug